ssdp.go

```diff
@@ -529,36 +530,42 @@ func (s *Ssdp) inMSearch(st string, req *http.Request, sendTo string) {
 	if st[0] == '"' && st[len(st)-1] == '"' {
 		st = st[1 : len(st)-2]
 	}
-	mx := 1
+	mx := time.Second
 	if mxStr := req.Header.Get("MX"); mxStr != "" {
 		mxInt64, err := strconv.ParseInt(mxStr, 10, 0)
 		if err != nil {
-			mx = int(mxInt64)
+			s.logger.Warnf("Could not parse mxStr: %s", err)
+			return
 		}
-		if mx < 1 {
-			mx = 1
-		} else if mx > 5 {
-			mx = 5
+		mx = time.Duration(mxInt64) * time.Second
+		if mx < time.Second {
+			mx = time.Second
+		} else if mx > 60*time.Second {
```

Comment:
I originally set this to 5 seconds to match the spec. A second reading shows that SHOULD rather than MUST be 5 seconds or less.

I'm fine with the new limit if you think it is still appropriate.

(spec for reference)

http://upnp.org/specs/arch/UPnP-arch-DeviceArchitecture-v1.1.pdf (page 31)
> MX
REQUIRED. Field value contains maximum wait time in seconds. MUST be greater than or equal to 1 and SHOULD be less than
5 inclusive. Device responses SHOULD be delayed a random duration between 0 and this many seconds to balance load for
the control point when it processes responses. This value MAY be increased if a large number of devices are expected to
respond. The MX field value SHOULD NOT be increased to accommodate network characteristics such as latency or
propagation delay (for more details, see the explanation below). Specified by UPnP vendor. Integer.

Reply:
I'm testing this with Amazon Alexa that sends MX of 15s, so I think larger value should be allowed with some reasonable limit. I think 1minute is a nice round limit.

Reply:
Sounds reasonable.